app/javascript/controllers/clipboard_controller.js

```diff
@@ -3,7 +3,18 @@ import { Controller } from "@hotwired/stimulus"
 export default class extends Controller {
   static values = { copy: String, tooltip: String }
 
+  originalStyleDisplay
+
   connect() {
+    this.originalStyleDisplay = this.element.style.display
+
+    navigator.permissions.query({ name: "clipboard-write" }).then((result) => {
+      this.updatePermissionState(result.state)
+      result.addEventListener("change", () => {
```

Comment:
So this is just checking if we have permission to write to the clipboard, and hiding the copy button entirely if we don't?  And how would it change, while the user is sitting on this page?

Does this event listener need to be disconnected at all?

Reply:
Actually, I do't think this needs cleaned up. This even handler is being attached to the permission query result, which happens once when navigating to that screen. When the user returns to this screen, a new query is done and the event is attached to the new result. Shouldn't have to worry about it firing multiple times.